srv/salt/state/conda_env_with_dask.sls

```diff
@@ -0,0 +1,33 @@
+# File:/srv/salt/conda_env_with_dask.sls
+
+check_conda_env_py39_dask:
+  cmd.run:
+    - name: 'conda env list | grep "py39_dask"'
+    - shell: /bin/bash
+    - env:
+      - PATH: /root/miniconda3/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/root/bin
```

Comment:
does this replace the path thats on the system? what if I have other PATH variables present on the system?

Reply:
I don't know. I presume so. 
The [salt cmd.run docs](https://docs.saltproject.io/en/latest/ref/states/all/salt.states.cmd.html) have:
```
{% set current_path = salt['environ.get']('PATH', '/bin:/usr/bin') %}

mycommand:
  cmd.run:
    - name: ls -l /
    - env:
      - PATH: {{ [current_path, '/my/special/bin']|join(':') }}
```
Which I believe would allow you to simply add something to PATH for the state to run. 
For what I am doing here, I don't know if overloading PATH, or adding to it would be universally helpful, in that I don't fully understand how order-precedence works with PATH. I can imagine a circumstance where simply augmenting PATH wouldn't use the path you're adding because another entry takes precedence.

Reply:
Interesting, when I use 
```
check_conda_env_py39_dask:
  cmd.run:
    - name: 'conda create --name py39_dask python=3.9'
    - shell: /bin/bash
    - env:
      - PATH: {{ [current_path, '/root/miniconda3/bin']|join(':') }}
    - require:
      - check_conda_install  # miniconda.sls; will ensure conda is set up.
```
The command finds conda ok, but seems to not find the previously created env: `                  WARNING: A conda environment already exists at '/root/miniconda3/envs/py39_dask'
                  Remove existing environment (y/[n])?` Which suggests to me that sourcing conda.sh directly is the better way to go.

Reply:
sourcing the conda file directly is... less elegant but I concede it might be better based on your findings.

Reply:
I think the better systemic approach might ultimately be to have some sort of env var with the path to the folder where Miniconda is to be installed - pass that to the installer, and also to `'source {{ CONDA_DIR }}/etc/profile.d/conda.sh` that I'm using to bypass any `.bashrc` issues. Unless you make that whole path an env var: 
```
CONDA_DIR=~/here
CONDA_SHELL=CONDA_DIR//etc/profile.d/conda.sh
```
That would also have the flexibility to have multiple installations of conda on the system, not that I can imagine much of a need for that, but still.